### PR TITLE
Skip upgrade review fixes

### PR DESCRIPTION
**Pull request by anilCSE**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added entries in `PENDING.md` with issue # 
- [ ] rereviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: simapp/test_helpers.go [hunk 1]
```diff
@@ -23,32 +23,37 @@ import (
 // Setup initializes a new SimApp. A Nop logger is set in SimApp.
 func Setup(isCheckTx bool) *SimApp {
 	db := dbm.NewMemDB()
-	app := NewSimApp(log.NewNopLogger(), db, nil, true, 0)
+	app := NewSimApp(log.NewNopLogger(), db, nil, true, []int64{}, 0)
```
Comment (ethanfrey) [inline simapp/test_helpers.go]: I was thinking the extra arg to `NewSimApp` would be a breaking change to many modules, but sure, other modules use `Setup` in test code. Seems fine.

File: x/upgrade/abci_test.go [hunk 2]
```diff
@@ -216,122 +238,145 @@ func (s *TestSuite) VerifyDone(newCtx sdk.Context, name string) {
...
 
-func (s *TestSuite) TestUpgradeSkippingOnlyTwo() {
-	newCtx := s.ctx.WithBlockHeight(s.ctx.BlockHeight() + 1).WithBlockTime(time.Now())
+func TestUpgradeSkippingOnlyTwo(t *testing.T) {
+	var (
+		skipOne   int64 = 11
+		skipTwo   int64 = 20
+		skipThree int64 = 25
+	)
+	s := setupTest(10, []int64{skipOne, skipTwo})
```
Comment (ethanfrey) [inline x/upgrade/abci_test.go]: Nice migrations to the non-suite form (above and below as well)

File: x/upgrade/abci_test.go [hunk 3]
```diff
@@ -216,122 +238,145 @@ func (s *TestSuite) VerifyDone(newCtx sdk.Context, name string) {
...
 }
 
-func (s *TestSuite) VerifySet() {
+func (s *TestSuite) VerifySet(skipUpgradeHeights []int64) {
 	s.T().Log("Verify if the skip upgrade has been set")
-	s.Require().NotNil(viper.GetIntSlice(s.FlagUnsafeSkipUpgrades))
+	s.Require().Equal(s.keeper.GetSkipUpgradeHeights(), skipUpgradeHeights)
 }
 
 func (s *TestSuite) VerifyConversion(skipUpgrade []int) {
```
Comment (ethanfrey) [inline x/upgrade/abci_test.go]: Can you change all these `s.VerifyXYZ(...)` methods to `VerifyXYZ(t, ...)` functions, using a normal `* testing.T`?

Once those are changed (and the calls to them), I think the rest of the `suite.Suite` references can be removed.

File: x/upgrade/abci_test.go [hunk 4]
```diff
@@ -17,6 +21,7 @@ import (
 )
 
 type TestSuite struct {
+	// TODO: remove this when we
```
Comment (ethanfrey) [inline x/upgrade/abci_test.go]: Let's finish this before merge

File: x/upgrade/abci_test.go [hunk 5]
```diff
@@ -27,333 +27,375 @@ type TestSuite struct {
 	FlagUnsafeSkipUpgrades string
 }
 
-func (s *TestSuite) SetupTest() {
+var s TestSuite
```
Comment (anilCSE) [inline x/upgrade/abci_test.go]: @ethanfrey do you suggest anything in place of this?

File: x/upgrade/abci_test.go [hunk 6]
```diff
@@ -49,104 +49,96 @@ func setupTest(height int64, skip []int64) TestSuite {
...
 	t.Log("Can overwrite plan")
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "bad_test", Height: s.ctx.BlockHeight() + 10}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 	err = s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Height: s.ctx.BlockHeight() + 1}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 
-	VerifyDoUpgrade(t)
+	s.VerifyDoUpgrade(t)
```
Comment (ethanfrey) [inline x/upgrade/abci_test.go]: Hey, that is a nice solution. Still keep the methods on `TestSuite` struct (as we need the keeper, etc), but no longer depend on the `suite.Suite` object exposing `.T()`. Nice fix up

**Pull request merged**